app/components/QrCodeScanner/QrCodeScanner.jsx

```diff
@@ -0,0 +1,66 @@
+// @flow
+import React, { Component } from 'react'
+
+import Button from '../Button'
+import Instascan from 'instascan'
+
+import qrCodeScannerStyles from './QrCodeScanner.scss'
+
+type Props = {
+  callback: Function,
+}
+
+type State = {
+  scanner: Object
+}
+
+class QrCodeScanner extends Component<Props, State> {
+  scanPreviewElement: HTMLVideoElement | null
+
+  state = {
+    scanner: {}
+  }
+
+  toggleScanner () {
+    const { scanner } = this.state
+
+    if (scanner.stop) {
+      scanner.stop()
+      this.setState(prevState => ({scanner: {}}))
+      return
+    }
+
+    const { callback } = this.props
+    const scannerInstance = new Instascan.Scanner({ video: this.scanPreviewElement })
+    scannerInstance.addListener('scan', function (content) {
+      scannerInstance.stop()
+      callback(content)
+    })
+    Instascan.Camera.getCameras().then(function (cameras) {
+      if (cameras.length > 0) {
```

Comment:
cameras will always be an array? if so, its better to type it

Reply:
Mention@evgenyboxer 🏁🚩☺